src/lib/server/tools/utils.ts

```diff
@@ -38,4 +41,17 @@ export async function callSpace<TInput extends unknown[], TOutput extends unknow
 		.then((res) => (res as unknown as GradioResponse).data as TOutput);
 }
 
+export async function getIpToken(sessionId: Conversation["sessionId"]) {
+	const session = await collections.sessions.findOne({ sessionId });
```

Comment:
No need for this fetch if the secret doesn't exist

Reply:
handled in ea7ad2451dc036e3d757017cd0778fde3f93f51a